src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -276,86 +279,87 @@ internal void Authenticate()
-                    var result = this.Send(workspace, workspaceProperties);
-                    var serverResponce = serverResponceRegex.Match(result);
-
-                    if (serverResponce.Success)
-                    {
-                        State = UploadState.Succeeded;
-                        Error = UploadErrorType.None;
-                        CustomizerURL = String.Concat(serverUrl, serverResponce.Value);
-                    }
-                    else if (InvalidNodeNames != null)
-                    {
-                        Error = UploadErrorType.InvalidNodes;
-                    }
-                    else
-                    {
-                        // If there wasn't any error during uploading, 
-                        // that means it's some error on the server side.
-                        Error = UploadErrorType.UnknownServerError;
-                    }
-                });
-
-        }
-
-        /// <summary>
-        /// Sends workspace and its' dependencies to Flood.
-        /// </summary>
-        /// <returns>String which is response from server.</returns>
-        internal string Send(HomeWorkspaceModel workspace, WorkspaceProperties workspaceProperties = null)
+        internal async void SendAsync(HomeWorkspaceModel workspace, WorkspaceProperties workspaceProperties = null)
         {
             if (String.IsNullOrWhiteSpace(serverUrl))
             {
                 Error = UploadErrorType.ServerNotFound;
-                return Resources.FailedMessage;
+                return;
             }
 
             if (String.IsNullOrWhiteSpace(authenticationProvider.Username))
             {
                 Error = UploadErrorType.AuthenticationFailed;
-                return Resources.FailedMessage;
+                return;
             }
 
-            if (authenticationProvider == null)
+            if (workspace.Nodes.Count() == 0)
             {
-                Error = UploadErrorType.AuthProviderNotFound;
-                return Resources.FailedMessage;
+                Error = UploadErrorType.EmptyWorkspace;
+                return;
             }
 
-            if (reachClient == null)
+            if (authenticationProvider == null)
             {
-                reachClient = new WorkspaceStorageClient(authenticationProvider, serverUrl);
+                Error = UploadErrorType.AuthProviderNotFound;
+                return;
             }
 
-            var dependencies = WorkspaceDependencies.Collect(workspace, customNodeManager);
+            State = UploadState.Uploading;
+
+            IReachHttpResponse result;
 
-            string result;
             try
             {
-                result = reachClient.Send(
-                    workspace,
-                    dependencies.CustomNodeWorkspaces,
-                    workspaceProperties);
-                InvalidNodeNames = null;
+                result = await Task.Run(() => this.Send(workspace, workspaceProperties));
```

Comment:
You can do this with  await Task.Run().ContinueWith - this gives a more controlled way to access the result message. 
 Task.Run(() => this.Send(workspace, workspaceProperties).ContinueWith(result => { 
   //check the status code and content here.
});